pkg/cassandra/version/version.go

```diff
@@ -0,0 +1,45 @@
+package version
+
+import (
+	"encoding/json"
+	"strconv"
+
+	"github.com/coreos/go-semver/semver"
+)
+
+type Version struct {
+	*semver.Version
+}
+
+func New(v string) *Version {
+	return &Version{semver.New(v)}
+}
+
+func (v *Version) Equal(versionB *Version) bool {
+	semverA := *v.Version
+	semverB := *versionB.Version
+	return semverA.Equal(semverB)
+}
+
+// UnmarshalJSON handles Cassandra not-quite-semver versions
+// which may not have a patch version.
+// See https://github.com/apache/cassandra/blob/cassandra-3.11.2/src/java/org/apache/cassandra/utils/CassandraVersion.java
+func (v *Version) UnmarshalJSON(data []byte) error {
+	if v.Version == nil {
+		v.Version = &semver.Version{}
+	}
+	s, err := strconv.Unquote(string(data))
+	if err != nil {
+		return err
+	}
+	err1 := v.Version.Set(s)
+	if err1 != nil {
+		err2 := v.Version.Set(s + ".0")
+		if err2 != nil {
+			return err1
+		}
```

Comment:
This is a very confusing dance of `err` vars. Can we flatten this to use `err` as much as possible, and if it isn't possible, give the second error a more helpful name than `err1`?